### ISSUE TITLE
JSON code block is not respecting line breaks in PDF output

### ISSUE THREAD

**Issue by yassersalama22** [ThomasLatham/markdown-pdf-plus#11]
**Description**:
When using the `markdown-pdf-plus` to convert a Markdown file into a PDF, I encountered an issue with code blocks. Specifically, when using a `json` code block in Markdown, like the following:

```json
{
  "name": "example",
  "version": "1.0.0",
  "dependencies": {
    "express": "^4.17.1"
  }
}
```

In the generated PDF, the code does not respect the line breaks from the source and is printed as a single line. Additionally, the code extends beyond the page boundaries, making it hard to read and damaging the layout of the PDF.

**Steps to Reproduce**:
1. Create a Markdown file containing a `json` code block with multiple lines.
2. Convert the Markdown file to PDF using `markdown-pdf-plus`.
3. Open the generated PDF to check the code block formatting.

**Expected Behavior**:
The code block should maintain the same line breaks and format as in the Markdown source. The text should also wrap or fit within the page boundaries.

**Actual Behavior**:
The entire JSON block appears on a single line in the PDF, and it extends beyond the page boundary, causing readability and layout issues.

**Environment**:
- markdown-pdf-plus version: v1.2.3
- OS: macOS 14.6.1

**Additional Information**:
This issue seems specific to the handling of code blocks, as regular text respects the line breaks. The issue occurs regardless of whether `json` or other code block languages are specified.

**Comment by ThomasLatham** [comment]
Thank you @yassersalama22 for raising this issue and for the very detailed write-up! I hope to find some time soon to look into what you've described and to implement a fix for it.

**Comment by ThomasLatham** [comment]
I've now reproduced the issue you described and am starting on implementing a fix. It seems that the changes I introduced in version 1.2.3 broke the display of code blocks in exported PDFs because the functionality is working in version 1.2.0.

**Comment by ThomasLatham** [comment]
Hi @yassersalama22, I've just published [version 1.2.4](https://github.com/ThomasLatham/markdown-pdf-plus/blob/main/CHANGELOG.md#124) to fix this issue. Can you please give this version a shot, and, if the issue is resolved on you end, please close this issue as completed? Thank you for your patience!
